### PR TITLE
CI: add mdformat

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

use MarkDown formatted, shall prevent eventual rending issues
no content changes...

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/basic_examples/README.md [hunk 1]
```diff
@@ -49,12 +61,14 @@ python autoencoder.py --trainer.gpus 2
...
 # Multi-node example
 
 This demo launches a job using 2 GPUs on 2 different nodes (4 GPUs total).
 To run this demo do the following:
 
 1. Log into the jumphost node of your SLURM-managed cluster.
-2. Create a conda environment with Lightning and a GPU PyTorch version.
-3. Choose a script to submit
+1. Create a conda environment with Lightning and a GPU PyTorch version.
+1. Choose a script to submit
```
Comment (awaelchli) [inline pl_examples/basic_examples/README.md:74]: what's happening here?

**Review by awaelchli — CHANGES_REQUESTED** [review]
something is going wrong with enumerated lists

File: pl_examples/basic_examples/README.md (hunk 1 above)
Comment (Borda) [reply pl_examples/basic_examples/README.md:74]: MD does not require any number, so in fact, if you write everywhere 9, you'll get the indexing from 1

**Comment by Borda** [comment]
> something is going wrong with enumerated lists

see https://guides.github.com/features/mastering-markdown/

**Review by akihironitta — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,94 +4,67 @@ All notable changes to this project will be documented in this file.
...
 
+*
 
 -
 
 ### Changed
 
 - Replace `iteration_count` and other index attributes in the loops with progress dataclasses ([#8477](https://github.com/PyTorchLightning/pytorch-lightning/pull/8477))
 
-
```
Comment (akihironitta) [inline CHANGELOG.md:27]: Is it okay that two empty lines between changelogs are now only one line?

Comment (Borda) [reply CHANGELOG.md:27]: well we can ignore chnagelog...

**Comment by Borda** [comment]
failing for https://github.com/executablebooks/mdformat/issues/247

**Review by ethanwharris — APPROVED** [review]
LGTM

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -311,54 +316,58 @@ def test_explain_what_is_being_tested(tmpdir):
     # BoringModel is a functional model. You might want to set methods to None to test your behaviour
     # Example: model.training_step_end = None
 
-    trainer = Trainer(default_root_dir=tmpdir, ...)  # will save everything within a tmpdir generated for this test
+    trainer = Trainer(
+        default_root_dir=tmpdir, ...
+    )  # will save everything within a tmpdir generated for this test
```
Comment (carmocca) [inline .github/CONTRIBUTING.md:321]: Does it set a line length? This is under 120 lines

**Review by carmocca — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (Borda) [reply .github/CONTRIBUTING.md:321]: good catch, opening issue https://github.com/hukkin/mdformat-black/issues/2

File: pl_examples/basic_examples/README.md (hunk 1 above)
Comment (awaelchli) [reply pl_examples/basic_examples/README.md:74]: that's just the strangest thing ever. 
markdown should be human readable in plain format, why would anyone write it like that and why would a formatter enforce it

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by edenlightning — APPROVED** [review]
